src/Thermal/RadwsEOS.cpp

```diff
@@ -82,7 +84,8 @@ FLOAT Radws<ndim>::EntropicFunction(Particle<ndim> &part)
 template <int ndim>
 FLOAT Radws<ndim>::SoundSpeed(Particle<ndim> &part)
 {
-  return sqrt(part.gamma*(part.gamma - 1.0)*part.u);
+  FLOAT gamma = opacity_table->GetGamma(part);
```

Comment:
I don't think this is the correct expression for the sound speed. While cs^2 = gamma P/ rho, P != (gamma-1) rho u for complicated equations of state. See D'Angelo et al. (2013, I think).

Can you also check that  the way gamma has been computed for the tables is also correct?

Reply:
Can you replace the first gamma with dlogP/dlogrho at constant entropy? The second one (gamma-1), is fine. 

If that's going to be a pain, then just replace the first one with 5/3. In SPH it doesn't matter because the sound speed doesn't enter the equations of motion, but it is important that our estimate for the sound speed is an upper bound (it's used for the Courant condition, where an over estimate is better than an under estimate).